Assets/Scripts/PlayerController/PlayerHandler.cs

```diff
@@ -0,0 +1,75 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class PlayerHandler : MonoBehaviour, ICanInteract
+{
+    [SerializeField]
+    private PlayerMovement m_PlayerMovement;
+
+    [SerializeField]
+    private PlayerAnimation m_PlayerAnimation;
+
+    [SerializeField]
+    private PowerupActor m_PowerupActor;
+
+    // Start is called before the first frame update
+    void Start()
+    {
+        m_PlayerAnimation = GetComponent<PlayerAnimation>();
+        m_PlayerMovement = GetComponent<PlayerMovement>();
+        m_PowerupActor = GetComponent<PowerupActor>();
+    }
+
+    // Update is called once per frame
+    void Update()
+    {
+        UpdateVelocity();
+    }
+
+    void OnTriggerEnter(Collider c)
+    {
+        IInteractable interactingObject = c.GetComponent<IInteractable>();
+        if (interactingObject != null)
+        {
+            interactingObject.Interact(this);
+        }
+    }
+
+    public void BoostJump() 
```

Comment:
Yeah, like terrence said, we don't want this code in the player handler class. It shouldn't know what the powerups do. The powerups should be the one applying the logic, and the playerhandler will only know that its something to interact with.

Reply:
Ok Sures will fix that in the next commit